scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -1326,30 +1563,32 @@ cdef void insert_node(FibonacciHeap* heap,
         heap.min_node = node
 
 
-cdef void decrease_val(FibonacciHeap* heap,
-                       FibonacciNode* node,
+cdef void decrease_val(FibonacciHeap * heap,
+                       FibonacciNode * node,
                        DTYPE_t newval):
     # Assumptions: - heap is a valid pointer
     #              - newval <= node.val
     #              - node is a valid pointer
     #              - node is not the child or sibling of another node
     #              - node is in the heap
     node.val = newval
+    if node.parent:
+        node.source = node.parent.source
     if node.parent and (node.parent.val >= newval):
         remove(node)
         insert_node(heap, node)
     elif heap.min_node.val > node.val:
         heap.min_node = node
 
 
-cdef void link(FibonacciHeap* heap, FibonacciNode* node):
+cdef void link(FibonacciHeap * heap, FibonacciNode * node):
     # Assumptions: - heap is a valid pointer
     #              - node is a valid pointer
     #              - node is already within heap
 
-    cdef FibonacciNode *linknode
-    cdef FibonacciNode *parent
-    cdef FibonacciNode *child
+    cdef FibonacciNode * linknode
```

Comment:
i'm sorry about this... i didn't realize that these autopep8 changes would be so dramatic.. i'll revert anf fix it.

Reply:
i've addressed this